src/site/content/en/blog/form-fatigue-validate-user-input/index.md

```diff
@@ -0,0 +1,308 @@
+
+No matter how optimized your forms are, users will make mistakes! When this
+happens, provide them with helpful feedback in real-time so they can quickly fix
+the issue. 
+
+## Restrict user input
+
+An effective way to avoid validation errors is to restrict user input (where
+possible) using built-in attributes.
+
++   **For numeric input**, use the `min` and `max` attributes to limit
+    input within a range. The `step` attribute can specify how much input
+    should each increment/decrement when adjusted by the slider or spinners.
+    For example, you can restrict a maximum price filter within a range of $50
+    to $500 with a `step` of 50:
+
+```html
+<input type="range" min="50" max="500" step="50">  
+```
+
+<!-- TODO (meggin):  image illustration or inline glitch? -->
+
++   **For text input**, use the `minlength` and `maxlength` attributes to
+    enforce limits on the length of text. For example, you can limit a
+    customized message on a flower delivery card to 50 characters:
+
+```html
+<input type="text" maxlength="50" >  
+```
+
+{% Aside 'caution' %}
+`minlength` and `maxlength` aren't enforced on mobile browsers. Always
+manually validate input on form submission.
+{% endAside %}
+
++   **For date and time input**, use `min` and `max` to restrict dates
+    and times. Browsers that provide a UI for choosing dates may prevent users
+    from entering invalid dates. For example, prevent users from selecting a
+    delivery date in the past or a delivery time outside of your operating hours:
+
+<figure class="w-figure">
+  <img src="image-browser-date-selector.png" alt="browser built-in date selector" style="max-width: 400px;">
+  <figcaption class="w-figcaption">
+    Chrome's default date picker.
+  </figcaption>
+</figure>
+
+
+```html
+<input type="date" min="2019-03-25" max="2019-04-25">  
+<input type="time" min="11:00" max="17:00">  
+```
+
+{% Aside 'caution' %}
+Browsers that do not support `date` or `time` fallback to type `text`. Always manually validate input on form submission.
+{% endAside %}
+
+## Use the `pattern` attribute
+
+In addition to restricting input, you can validate input against a regular
+expression using the `pattern` attribute. For example, a valid US zip code is 5
+digits and sometimes followed by a dash and an additional 4 digits:
+
+
+```html  
+<input type="text" pattern="^\d{5,6}(?:[-\s]\d{4})?$" ...>  
+```
+
+The following regular expression validates credit card numbers:
+
+
+```html
+^(?:4[0-9]{12}(?:[0-9]{3})?|5[1-5][0-9]{14}|3[47][0-9]{13}|3(?:0[0-5]|[68][0-9])[0-9]{11}|6(?:011|5[0-9]{2})[0-9]{12}|(?:2131|1800|35\d{3})\d{11})$  
+```
+
+{% Aside 'note' %}
+Unlike `pattern`, `min`, `max`, `minlength` and `maxlength` will restrict user input in real-time.
+{% endAside %}
+
+## Custom validation
+
+If built-in validation attributes aren't enough, use the
+[Constraint Validation API](https://w3c.github.io/html/sec-forms.html#constraints)
```

Comment:
I'd recommend linking to one of the MDN references on it as they might be easier for folks to parse.

https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5/Constraint_validation

or

https://developer.mozilla.org/en-US/docs/Web/API/Constraint_validation

could be good.